Occasional segfault in `KernelContextException.test.cc`

Reported by @pcanal.
<details><summary>segfault details</summary><pre>
80: info: Loading VecGeom geometry from GDML at /nashome/p/pcanal/geant/sources/celeritas/test/celeritas/data/two-boxes.gdml
80: status: Initializing tracking information
80: /nashome/p/pcanal/geant/sources/celeritas/src/celeritas/phys/PhysicsParams.cc:300: warning: No processes are defined for particle 'electron'
80: [       OK ] KernelContextExceptionTest.typical (38 ms)
80: [ RUN      ] KernelContextExceptionTest.uninitialized_track
80: /nashome/p/pcanal/geant/sources/celeritas/src/celeritas/phys/PhysicsParams.cc:300: warning: No processes are defined for particle 'electron'
80:
80:  *** Break *** segmentation violation
80:
80:
80:
80: ===========================================================
80: There was a crash.
80: This is the entire stack trace of all threads:
80: ===========================================================
80: #0  0x00007f087dbce60c in waitpid () from /lib64/libc.so.6
80: #1  0x00007f087db4bf62 in do_system () from /lib64/libc.so.6
80: #2  0x00007f087465808c in TUnixSystem::StackTrace() () from /nashome/p/pcanal/geant/gcc10/install/opt/lib/libCore.so
80: #3  0x00007f08746557a5 in TUnixSystem::DispatchSignals(ESignals) () from /nashome/p/pcanal/geant/gcc10/install/opt/lib/libCore.so
80: #4  <signal handler called>
80: #5  0x00007f08819d564a in vecgeom::cxx::VPlacedVolume::GetLogicalVolume (this=0x0) at /nashome/p/pcanal/geant/gcc10/install/scalar-cuda11/VecGeom/include/VecGeom/volumes/PlacedVolume.h:152
80: #6  0x00007f08819d59b8 in celeritas::VecgeomTrackView::volume (this=0x7ffea089a810) at /nashome/p/pcanal/geant/sources/celeritas/src/celeritas/ext/VecgeomTrackView.hh:487
80: #7  0x00007f08819d4e64 in volume_id (this=0x7ffea089a810) at /nashome/p/pcanal/geant/sources/celeritas/src/celeritas/ext/VecgeomTrackView.hh:238
80: #8  celeritas::KernelContextException::initialize (this=0x7ffea089b490, core=...) at /nashome/p/pcanal/geant/sources/celeritas/src/celeritas/global/KernelContextException.cc:147
80: #9  0x00007f08819d45bd in celeritas::KernelContextException::KernelContextException(celeritas::CoreRef<(celeritas::MemSpace)0> const&, celeritas::OpaqueId<celeritas::Thread, unsigned int>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&) (this=0x7ffea089b490, data=..., thread=..., label=<unknown type in /nashome/p/pcanal/geant/gcc10/builds/scalar-cuda11/celeritas/lib64/libceleritas.so, CU 0x74ca62, DIE 0x7a82bb>) at /nashome/p/pcanal/gean
</pre></details>

This I think is just the KCE trying to be too aggressive about querying uninitialized states.

--- a/src/celeritas/global/KernelContextException.cc
+++ b/src/celeritas/global/KernelContextException.cc
@@ -64,7 +64,7 @@ KernelContextException::KernelContextException(CoreHostRef const& data,
         if (!thread)
         {
             // Make sure the thread is valid before trying to construct
-            // detailed debug informatino
+            // detailed debug information
             throw std::exception();
         }
         CoreTrackView core(data.params, data.states, thread);
@@ -128,25 +128,26 @@ void KernelContextException::output(JsonPimpl* json) const
  */
 void KernelContextException::initialize(CoreTrackView const& core)
 {
+    auto const&& sim = core.make_sim_view();
+    if (sim.status() == TrackStatus::alive)
     {
-        auto const&& sim = core.make_sim_view();
         event_ = sim.event_id();
         track_ = sim.track_id();
         parent_ = sim.parent_id();
         num_steps_ = sim.num_steps();
-    }
-    {
-        auto const&& par = core.make_particle_view();
-        particle_ = par.particle_id();
-        energy_ = par.energy();
-    }
-    {
-        auto const&& geo = core.make_geo_view();
-        pos_ = geo.pos();
-        dir_ = geo.dir();
-        volume_ = geo.volume_id();
-        surface_ = geo.surface_id();
-        next_surface_ = geo.next_surface_id();
+        {
+            auto const&& par = core.make_particle_view();
+            particle_ = par.particle_id();
+            energy_ = par.energy();
+        }
+        {
+            auto const&& geo = core.make_geo_view();
+            pos_ = geo.pos();
+            dir_ = geo.dir();
+            volume_ = geo.volume_id();
+            surface_ = geo.surface_id();
+            next_surface_ = geo.next_surface_id();
+        }
     }
     {
         // Construct std::exception message

--- a/test/celeritas/global/KernelContextException.test.cc
+++ b/test/celeritas/global/KernelContextException.test.cc
@@ -155,12 +155,9 @@ TEST_F(KernelContextExceptionTest, uninitialized_track)
     step(make_span(primaries));
 
     this->check_kce = [](KernelContextException const& e) {
-        if (!CELERITAS_USE_VECGEOM)
-        {
-            // Don't test this with vecgeom which has more assertions when
-            // acquiring data
-            EXPECT_STREQ("kernel context: thread 1 in 'test-kernel'", e.what());
-        }
+        // Don't test this with vecgeom which has more assertions when
+        // acquiring data
+        EXPECT_STREQ("kernel context: thread 1 in 'test-kernel'", e.what());
         EXPECT_EQ(ThreadId{1}, e.thread());
         EXPECT_EQ(EventId{}, e.event());
         EXPECT_EQ(TrackId{}, e.track());

